### DISCUSSION TITLE
notification: filter done/undone

### DISCUSSION THREAD

**Discussion by antonengelhardt** [community/community#118736] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

Hello,

the [GitHub Raycast Extension](https://www.raycast.com/raycast/github) has a command (`Notifications`) to fetch all notifications for the user using the `listNotificationsForAuthenticatedUser()` endpoint:

![Screenshot-Arc-006646@2x](https://github.com/octokit/octokit.js/assets/106314688/f153fb43-7315-49e4-9aee-62c17f9a1bc3)

```ts
const {
    data,
    isLoading,
    mutate: mutateList,
  } = useCachedPromise(async () => {
    const response = await octokit.rest.activity.listNotificationsForAuthenticatedUser({ all: true });
    return response.data;
  });
```

When setting the `all`-parameter to `true`, it returns all notifications (including done ones). When setting it to `false`, only unread notifications are returned.

The GitHub Website groups notifications depending if they're done or undone. Additionally, a filter can display only the unread notifications: 

![Screenshot-Arc-006648@2x](https://github.com/octokit/octokit.js/assets/106314688/cc81cb73-fbe7-470e-b45f-68426b8c1e26)

@tom-sherman proposed to follow the GitHub style in a recently opened [issue](https://github.com/raycast/extensions/issues/11751). Following the Github style would mean that the extension would have to group the notifications by undone/done and highlight unread ones, which is currently impossible, because there is no way to tell if a notification is done or not.

My proposal is that either
1. the notification object gets a `done` attribute of `boolean` type (then we can partition the notifications ourselves) or 
2. that the `listNotificationsForAuthenticatedUser()` [endpoint](https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#list-notifications-for-the-authenticated-user) gets more filters e.g `showDone` (`bool`), `showRead` (`boolean`).

Thanks for considering. Cheers.

### SDK Version

v3.1.2 Octokit js

**Comment by mkurz** [comment]
In case the done will get returned from the api, please make it a timestamp, when not null it should contain the moment when a notification was "clicked" as done. Thanks!
